### PR TITLE
Move RunResult to schema and update command signature

### PR DESCRIPTION
**Pull request by kjanat**
### Schema Architecture Refactoring

- **Move `RunResult` to schema layer**: `RunResult` type relocated from `testkit/index.ts` to dedicated `src/core/schema/run.ts`, with `testkit/index.ts` re-exporting for backward compatibility
- **Relax command options typing**: `ErasedCommand._execute()` now accepts `Readonly<Record<string, unknown>>` for maximum flexibility at the schema boundary (previously `RunOptions`)
- **Expand public API surface**: `RunResult` exported from `src/core/schema/index.ts` and included in top-level exports from `src/index.ts`

### Package.json Auto-Discovery

- **Add `CLIBuilder.packageJson(settings?)`**: Opt-in auto-discovery of package.json with optional `inferName` setting
- **Add `discoverPackageJson(adapter)`**: Walk up directory tree to locate and parse nearest package.json, returning metadata (name, version, description, bin)
- **Add `inferCliName(pkg)`**: Infer CLI name from package.json with precedence: bin object first key → package name (with scope stripped)
- **Intelligent merging**: Discovered metadata (version, description, inferred name) merged into CLI schema, respecting explicit values and skipping during completions

### Default Command Support

- **Add `CLIBuilder.default(cmd)`**: Register a single default command invoked for empty argv or unknown commands
- **Smart dispatch**: Default command participates in command resolution; unknown args proxy to default when appropriate
- **Help integration**: Root help displays default command with " (default)" tag; usage line shows optional `[command]` instead of required `<command>`

### Environment Variable Argument Resolution

- **Add `ArgBuilder.env(varName)`**: Bind environment variable to positional argument
- **Coercion & resolution**: Environment values coerced to arg kind with precedence: CLI → env → default
- **Enhanced help**: Arguments with env vars annotated with `[env: VAR]` in help output
- **Error handling**: Type mismatch and missing required arg errors include helpful suggestions for env-configured args

### Documentation & Tooling

- **Comprehensive JSDoc coverage**: Expanded documentation for `ArgBuilder`, `ArgFactory`, `CommandBuilder`, and related APIs with examples
- **Build system updates**: 
  - TypeScript: added `isolatedDeclarations`, `declaration`, `declarationMap` flags
  - Tsdown: restructured format/dts/exports/target configs with per-target settings
  - Deno: replaced `deno.json` with `deno.jsonc` including tasks, refined publish config
  - Added `bunfig.toml` for bun runtime configuration with version pinning
- **Editor integration**: Zed settings now use simplified dprint invocation with Deno LSP support
- **Dprint**: Added JSON trailing comma configuration for VSCode/Zed settings and tsconfig files

### Package & Dependencies

- Version bumped: 0.9.0 → 0.9.1
- Added runtime-specific exports: bun entry points mirroring npm exports
- Added type definitions: `@types/bun`, `@types/node`
- Updated TypeScript to ^5.9.3 and `@typescript/native-preview`
- Engines expanded: require bun ≥1.3, deno ≥2.6, node ≥22

### Testing

Added comprehensive test suites:
- CLI default command behavior & dispatch
- Package.json discovery across directory trees with edge cases
- CLI name inference from package.json with scoped package handling
- Argument environment variable resolution with precedence validation
- Help command virtual subcommand integration
- Nested help interactions
- JSON mode propagation with help

### Lines Changed
- Total: ~3000+ lines across 40+ files
- Primary effort: schema refactoring, new feature implementations, extensive test coverage

### CODE REVIEW

**Comment by kjanat** [comment]
@coderabbitai help

**Comment by kjanat** [comment]
@coderabbitai generate configuration

**Comment by kjanat** [comment]
@coderabbitai full review

**Review by kjanat — COMMENTED** [review]
👍

File: src/core/cli/cli-default.test.ts [hunk 1]
```diff
@@ -0,0 +1,310 @@
...
+import { describe, expect, it, vi } from 'vitest';
+import { CLIError } from '../errors/index.ts';
+import { arg } from '../schema/arg.ts';
+import { command } from '../schema/command.ts';
+import { flag } from '../schema/flag.ts';
+import { cli, formatRootHelp } from './index.ts';
+
+// ===================================================================
+// Helpers
+// ===================================================================
```
[the remark this answers is not in the document]
Comment (kjanat) [reply src/core/cli/cli-default.test.ts:14]: `.test.ts`...

File: tsdown.config.ts [hunk 2]
```diff
@@ -6,16 +6,29 @@ export default defineConfig({
...
-	exports: true,
-	minify: true,
+	exports: {
+		devExports: 'bun',
+		legacy: true,
+		enabled: true,
+		packageJson: true,
+	},
+	unbundle: true,
+	minify: 'dce-only',
```
[the remark this answers is not in the document]
Comment (kjanat) [reply tsdown.config.ts:31]: Your mother is not a valid field

File: src/core/resolve/resolve-arg-env.test.ts [hunk 3]
```diff
@@ -0,0 +1,347 @@
+import { describe, expect, it } from 'vitest';
+import { isValidationError, type ValidationError } from '../errors/index.ts';
+import type { ParseResult } from '../parse/index.ts';
+import { createArgSchema } from '../schema/arg.ts';
+import type { CommandArgEntry, CommandSchema } from '../schema/command.ts';
+import type { ResolveOptions } from './index.ts';
+import { resolve } from './index.ts';
```
[the remark this answers is not in the document]
Comment (kjanat) [reply src/core/resolve/resolve-arg-env.test.ts:7]: @coderabbitai `.test.ts` again... pls close all comments asking to add `@module` JSDoc to test files

**Comment by kjanat** [comment]
@coderabbitai Is #7 a superset of the changes in this pr? Would it benefit from doing whatever this pr aimed to do, or should I drop this PR?

**Comment by kjanat** [comment]
Covered in PR #7.

See https://github.com/kjanat/dreamcli/pull/4#issuecomment-4164113116.
